### PR TITLE
[DO NOT MERGE] Testing new proxy build

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

**Comment by nmittler** [comment]
@ldemailly any idea why prow seems [unable to download my image](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/3690/istio-presubmit/5397/)? I've made the gcs bucket readable for all authenticated users and verified I can access it from another account.

File: bin/init.sh [hunk 1]
```diff
@@ -42,55 +42,94 @@ if [ ${ROOT} != "${GO_TOP:-$HOME/go}/src/istio.io/istio" ]; then
...
-    echo "Downloading envoy $PROXY using $DOWNLOAD_COMMAND"
-    time ${DOWNLOAD_COMMAND} https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xz
-    cp usr/local/bin/envoy $OUT/envoy-$PROXYVERSION
+    # Pull the image to the local docker registry.
+    DOWNLOAD_COMMAND="docker pull $PROXY_DOCKER_IMAGE"
+    echo "Downloading envoy using $DOWNLOAD_COMMAND"
+    ${DOWNLOAD_COMMAND}
+
+    # Run the image and pull out the envoy executable
+    ENVOY_EXTRACT_COMMAND="docker run $PROXY_DOCKER_IMAGE cat /usr/local/bin/envoy > $PROXY_EXTRACTED_BIN_NAME"
```
Comment (ldemailly) [inline bin/init.sh]: please be mindful, this ends up killing prow by cat'ing large binary into the log
https://storage.googleapis.com/istio-prow/pull/istio_istio/3690/istio-presubmit/5550/build-log.txt

**Comment by nmittler** [comment]
No longer need this test, since istio/istio no longer depends on istio/proxy dockerfiles.

**Pull request closed without merging**